lib/isa_exporter.rb

```diff
@@ -585,6 +640,18 @@ def group_samples_by_input_and_parameter_value(sample_type)
       grouped_samples.transform_keys { |key| group_id(key) }
     end
 
+    def process_sequence_input(inputs, type)
+      input_ids = inputs.map { |input| input[:id] }
+      authorized_sample_ids = Sample.where(id: input_ids).select { |s| s.can_view?(@current_user) }.map(&:id).compact
```

Comment:
you could use ...`Sample.where(id: input_ids).authorized_for(:view, @current_user)`... which will filter in a single query. Might not matter here, but can make a difference where there are large number and avoids looping over.
Not a blocker though